### PR TITLE
Feature/otp

### PR DESCRIPTION
**Pull request by deltaDirk**
**Type**
Feature
**Description**
All the changes needed to facilitate the new OTP flow
**Related Tickets**
COV-283

### CODE REVIEW

File: Migrations/20200507151313_TestTypeToTestResult.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace CoviIDApiCore.Migrations
+{
+    public partial class TestTypeToTestResult : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.AddColumn<int>(
+                name: "TestType",
```
Comment (Roleen20) [inline Migrations/20200507151313_TestTypeToTestResult.cs:10]: Should that not be an enum reading as string, instead of an int?

File: Startup.cs [hunk 2]
```diff
@@ -176,6 +176,7 @@ private void ConfigureDependecyInjection(IServiceCollection services)
             services.AddScoped<ICovidTestRepository, CovidTestRepository>();
             services.AddScoped<IWalletDetailRepository, WalletDetailRepository>();
             services.AddScoped<IWalletTestResultRepository, WalletTestResultRepository>();
+            services.AddScoped<IWalletDetailService, WalletDetailService>();
```
Comment (Roleen20) [inline Startup.cs:179]: under the service region

File: Migrations/20200507151313_TestTypeToTestResult.cs (hunk 1 above)
Comment (deltaDirk) [reply Migrations/20200507151313_TestTypeToTestResult.cs:10]: Ahh you're right, forgot the conversion, will add it in

File: Startup.cs (hunk 2 above)
Comment (deltaDirk) [reply Startup.cs:179]: Oops, thanks

File: V1/DTOs/WalletTestResult/TestResultRequest.cs [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
     public class TestResultRequest
     {
         public string Key { get; set; }
-        public LaboratoryStatus LaboratoryStatus { get; set; }
+        public LaboratoryStatus LaboratoryStatus { get; set; } = LaboratoryStatus.Unsent;
         public ResultStatus ResultStatus { get; set; }
         public Laboratory Laboratory { get; set; }
         public string ReferenceNumber { get; set; }
         public DateTime TestedAt { get; set; }
+        public DateTime IssuedAt { get; set; }
```
Comment (Roleen20) [inline V1/DTOs/WalletTestResult/TestResultRequest.cs:13]: This is an internal field. It only gets added to the database when the test results have been verified with a lab

File: V1/DTOs/WalletTestResult/TestResultRequest.cs [hunk 4]
```diff
@@ -10,7 +10,11 @@ public class TestResultRequest
         public Laboratory Laboratory { get; set; }
         public string ReferenceNumber { get; set; }
         public DateTime TestedAt { get; set; }
-        public DateTime IssuedAt { get; set; }
         public bool HasConsent { get; set; }
+
+        public bool isValid()
+        {
+            return TestedAt.Date > DateTime.UtcNow.Date;
```
Comment (Roleen20) [inline V1/DTOs/WalletTestResult/TestResultRequest.cs:17]: should be smaller than today?

**Pull request merged**